wagtail/admin/views/generic/models.py

```diff
@@ -398,6 +398,11 @@ def get_add_url(self):
             )
         return reverse(self.add_url_name)
 
+    def get_edit_url(self):
+        if self.edit_url_name is None:
+            return None
```

Comment:
As far as I can see we don't have any meaningful fallback behaviour in place for when this returns None, so it probably makes sense for this to raise an ImproperlyConfigured exception like the other `get_*_url` methods.

While we're at it, `get_success_buttons` could also be updated to call `get_edit_url`.

Reply:
Ah, you're right. I was hesitant to add the fallback as I thought the generic view itself doesn't really use the `edit_url_name`, but now I see that the logic is already in `get_success_buttons`.

Done, thanks!